### PR TITLE
Fix #5485 Create means for verifying Fragment Arguments

### LINKED ISSUE #5485: [Feature Request]: Create means for verifying Fragment Arguments
### Is your feature request related to a problem? Please describe.

We use Espresso `Intents.Intended` to verify activity intent params, but we also need to test the fragment arguments and saved state bundles for correcteness, and to prevent bugs during development.

### Describe the solution you'd like

An example implementation was provided in [Fix part of #5485: Create means for verifying Fragment Arguments #5606](https://github.com/oppia/oppia-android/pull/5606). 

### Describe alternatives you've considered
Not testing fragement arguments, but since we are working long term on improving coverage, and also wanting to catch regressions, this is not an option.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by subhajitxyz**
## Explanation

   Feature Request part1 #5485
   Added test for 10 fragments arguments and saveInstanceState.
 

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- For PRs introducing new UI elements or color changes, both light and dark mode screenshots must be included
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by theMr17** [comment]
@subhajitxyz per discussion through chat, some checks were canceled, possibly due to closing and reopening the PR. I have re-run the checks, and everything seems to be fine now.

**Comment by subhajitxyz** [comment]
Thanks @theMr17 .

**Comment by subhajitxyz** [comment]
@adhiamboperes, PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedFragmentTest.kt [hunk 1]
```diff
@@ -901,6 +904,70 @@ class MarkChaptersCompletedFragmentTest {
     }
   }
 
+  @Test
+  fun testMarkChaptersCompletedFragment_arguments_workingProperly() {
```
Comment (adhiamboperes) [inline app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedFragmentTest.kt]: Rename this and other tests to be more descriptive.
```suggestion
  fun testFragment_fragmentLoaded_verifyCorrectArgumentsPassed() {
```

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @subhajitxyz! 

The implementation looks good to. My only suggestion is to refactoor the test names to follow the pattern in my suggestion.

File: app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedFragmentTest.kt (hunk 1 above)
Comment (subhajitxyz) [reply app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedFragmentTest.kt]: Ok

Comment (subhajitxyz) [reply app/src/sharedTest/java/org/oppia/android/app/devoptions/MarkChaptersCompletedFragmentTest.kt]: Done.

**Comment by subhajitxyz** [comment]
Hi @theMr17, Could you please tell me why this test failed ?ExplorationActiveTimeControllerTest > testSessionTimer_explorationStarted_appInBackground_pausesSessionTimer FAILED

**Comment by theMr17** [comment]
> Hi @theMr17, Could you please tell me why this test failed ?ExplorationActiveTimeControllerTest > testSessionTimer_explorationStarted_appInBackground_pausesSessionTimer FAILED

@subhajitxyz The test seems to be flaky, I have re-run the test, hopefully the check will pass this time.

/cc @adhiamboperes.

**Comment by subhajitxyz** [comment]
@adhiamboperes , PTAL.

**Review by adhiamboperes — APPROVED** [review]
Thanks @subhajitxyz! This LGTM.

**Pull request merged**